dotnet/hello_world/Program.cs

```diff
@@ -0,0 +1,210 @@
+
+                    Console.WriteLine($"Writing some greetings to the table {_settings.TableName}");
+
+                    //string[] greetings = new string[] { "Hello World!", "Hellow Bigtable!", "Hellow C#!" };
+
+                    BigtableClient bigtableClient = BigtableClient.Create();
+
+                    //building a MutateRowsRequest (contains table name and collection of entries)
+                    MutateRowsRequest request = new MutateRowsRequest
+                    {
+                        TableNameAsTableName = _tableClient
+                    };
+
+                    for (int i = 0; i < _settings.Greetings.Length; i++)
+                    {
+                        string rowKey = String.Format("{0}{1}", _settings.RowKeyPrefix, i);
+
+                        //Building an entry for every greeting (contains of row key and collection of mutations)
+                        MutateRowsRequest.Types.Entry _entry = new Google.Cloud.Bigtable.V2.MutateRowsRequest.Types.Entry();
+                        _entry.Mutations.Add(Mutations.SetCell(_settings.ColumnFamily, _settings.ColumnName, new BigtableByteString(_settings.Greetings[i])));
+                        _entry.RowKey = ByteString.CopyFromUtf8(rowKey);
+
+                        //adding entry to request
+                        request.Entries.Add(_entry);
+                    }
+
+                    BigtableClient.MutateRowsStream response = bigtableClient.MutateRows(request);
+
+                    Task write = CheckWrittenAsync();
+                    write.Wait();
+
+                    async Task CheckWrittenAsync()
+                    {
+                        IAsyncEnumerator<MutateRowsResponse> responseStream = response.ResponseStream;
+                        while (await responseStream.MoveNext())
+                        {
+                            MutateRowsResponse current = responseStream.Current;
+                        }
+                    }
+                    #endregion
+
+                    #region reading the first row
+                    Console.WriteLine("Reading the first row");
+                    int number = 1;
+                    for (int i = 0; i < number; i++)
+                    {
+                        BigtableByteString rowKey = String.Format("{0}{1}", _settings.RowKeyPrefix, i);
+                        Row row = bigtableClient.ReadRow(_tableClient, rowKey);
+
+                        Console.WriteLine($"\tRow key: {row.Key.ToStringUtf8()}, Value: {row.Families[0].Columns[0].Cells[0].Value.ToStringUtf8()}");
+                    }
+                    #endregion
+
+                    #region reading all rows
+
+                    BigtableClient.ReadRowsStream responseRead = bigtableClient.ReadRows(new ReadRowsRequest
+                    {
+                        TableNameAsTableName = new Google.Cloud.Bigtable.V2.TableName(ProjectId, InstanceId, _settings.TableName)
```

Comment:
`TableNameAsTableName` is a bit confusing as a name. What does it mean? What does it do? How's this used?

Reply:
TableNameAsTableName is a name of the property in [ReadRowsRequest][ReadRowsRequest]. Same is in [MutateRowsRequest][MutateRowsRequest] and all other requests

[ReadRowsRequest]:https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/315bea078fdc376fb7b2c8263a25e0d3f6ebd7e8/apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/ResourceNames.cs#L180

[MutateRowsRequest]:https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/315bea078fdc376fb7b2c8263a25e0d3f6ebd7e8/apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/ResourceNames.cs#L154